app/src/main/java/com/example/taskthree/view/GuessFragment.kt

```diff
@@ -0,0 +1,83 @@
+package com.example.taskthree.view
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.fragment.findNavController
+import com.example.taskthree.databinding.FragmentCharBinding
+import com.example.taskthree.viewmodel.GuessViewModel
+
+class GuessFragment : Fragment() {
+    private lateinit var binding: FragmentCharBinding
+    private val viewModel: GuessViewModel by activityViewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        binding = FragmentCharBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        buttonConnectEditText()
+        viewModel.resultMessageObs.observe(viewLifecycleOwner) { message ->
+            if (message == TRY_AGAIN_MESSAGE) {
+                binding.textViewGuessHint.text = viewModel.randomCharObs.toString()
+            } else if (message == WIN_MESSAGE) {
+                binding.textViewGuessHint.text =
+                    StringBuilder().append(GUESS_CHARACTER_HINT + viewModel.randomIntObs.value.toString())
```

Comment:
GUESS_CHARACTER_HINT strings.xml'den aldiginda StringBuilder yerine formatlamayi deneyebiliriz. Ornek;
<string name="welcome_messages">Hello, %1$s! You have %2$d new messages.</string>
var text = getString(R.string.welcome_messages, username, mailCount)